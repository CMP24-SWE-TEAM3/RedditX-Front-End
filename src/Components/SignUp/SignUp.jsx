// imports
import React from 'react';
import {SignUpStyled} from "Components/SignUp/SignUp.styled";

/**
 * Component that displays SignUp button in Navbar'
 * @returns {React.Component}
 */
const SignUp = () => {
    const alertSignUp = () => {
        alert("Welcome on Board!");
    }
    return (
        <SignUpStyled id={'sign-up-button-from-HomePage'} onClick={alertSignUp} variant="outline-primary rounded-pill h-4">Sign Up</SignUpStyled>
    );
};

export default SignUp;
