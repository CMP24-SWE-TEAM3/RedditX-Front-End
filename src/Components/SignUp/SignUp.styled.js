import styled from "styled-components";
import {Button} from "react-bootstrap";

export const SignUpStyled = styled(Button)`
  //color: #1779d3;
  color:${({theme})=>theme.color.secondary};
  border-color: ${({theme})=>theme.color.secondary};
  width: 118px;
  height: 32px;
  padding: 0 15px;
  margin-left: 5%;
  font-size: 14px;

  &:hover {
    color: #1779d3 !important;
    background-color: #f5fafd !important;
  }

  &:active {
    color: #1779d3 !important;
    background-color: #ebf4fc!important;
  }

  font-weight: bold;
  //height: min-content;
`