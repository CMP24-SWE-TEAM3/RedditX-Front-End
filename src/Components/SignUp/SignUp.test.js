import {shallow} from 'enzyme';

// import Components
import SignUp from "./SignUp";

describe("Sign Up Component", () => {
    it("this is a test for Sign Up Button", () => {
        expect(shallow(<SignUp/>)).toMatchSnapshot();

    });
});