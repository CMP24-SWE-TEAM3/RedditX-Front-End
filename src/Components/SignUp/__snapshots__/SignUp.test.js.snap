// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Component this is a test for Sign Up Button 1`] = `ShallowWrapper {}`;
